src/debugger/gdb-stub.c

```diff
@@ -433,7 +447,62 @@ static void _processQSupportedCommand(struct GDBStub* stub, const char* message)
 		}
 		message = end + 1;
 	}
-	strncpy(stub->outgoing, "swbreak+;hwbreak+", GDB_STUB_MAX_LINE - 4);
+	strncpy(stub->outgoing, "swbreak+;hwbreak+;qXfer:features:read+;qXfer:memory-map:read+", GDB_STUB_MAX_LINE - 4);
+}
+
+static void _processQXferCommand(struct GDBStub* stub, const char* params, const char* data) {
+	unsigned offset = 0;
+	unsigned length = 0;
+
+	unsigned index = 0;
+	for (index = 0; params[index] != ','; ++index) {
+		offset <<= 4;
+		offset |= _hex2int(&params[index], 1);
+	}
+
+	++index;
+	int lengthStrParams = strlen(params);
+	for (; index + 3 < lengthStrParams; ++index) {
+		length <<= 4;
+		length |= _hex2int(&params[index], 1);
+	}
+
+	if (length + 1 > GDB_STUB_MAX_LINE) {
+		length = GDB_STUB_MAX_LINE - 1;
+	}
+
+	memset(stub->outgoing, 0, GDB_STUB_MAX_LINE);
+	if (strlen(data) < length + offset) {
+		length = strlen(data) - offset;
+		stub->outgoing[0] = 'l';
+	} else {
+		stub->outgoing[0] = 'm';
+	}
+	strncpy(&stub->outgoing[1], &data[offset], length);
```

Comment:
This is a very specific edge case, but: use `strlcpy` instead of `strncpy`; That will fix the edge case where the size of the buffer and the size of the data are equal--in the current case, the null terminator would get cropped out. strlcpy handles this properly.

Reply:
done...but for some reason, I had to do + 1 on the length to make it not truncate the last character? that's odd.

Reply:
That's a bad sign...means you're computing the maximum length wrong somewhere. I'll take a check tomorrow to see if I can figure it out.